core/src/main/java/org/eclipse/hono/util/ResourceIdentifier.java

```diff
@@ -144,49 +82,62 @@ private String createStringRepresentation(final int startIdx) {
         return b.toString();
     }
 
-    private void createStringRepresentation() {
-        resource = createStringRepresentation(0);
-
-        final StringBuilder b = new StringBuilder(getEndpoint());
-        if (getTenantId() != null) {
-            b.append("/").append(getTenantId());
-        }
-        basePath = b.toString();
+    /**
+     * Checks whether the given string is a valid resource identifier string representation.
+     *
+     * @param resource the resource string to parse.
+     * @return {@code true} if the given string is a valid resource identifier string.
+     */
+    public static boolean isValid(final String resource) {
+        return !Strings.isNullOrEmpty(resource) && !resource.startsWith("/");
     }
 
     /**
      * Creates a resource identifier from its string representation.
      * <p>
      * The given string is split up into segments using a forward slash as the separator. The first segment is used as
      * the endpoint, the second segment is used as the tenant ID and the third segment (if present) is used as the
-     * device ID.
+     * resourceId part.
      * </p>
      *
      * @param resource the resource string to parse.
      * @return the resource identifier.
      * @throws NullPointerException if the given string is {@code null}.
-     * @throws IllegalArgumentException if the given string is empty.
+     * @throws IllegalArgumentException if the given string is empty or contains an empty first segment.
      */
     public static ResourceIdentifier fromString(final String resource) {
         Objects.requireNonNull(resource);
-        if (resource.isEmpty()) {
-            throw new IllegalArgumentException("resource must not be empty");
-        }
-        return new ResourceIdentifier(resource);
+        final String[] path = resource.split("/");
+        final List<String> pathSegments = new ArrayList<>(Arrays.asList(path));
+        return new ResourceIdentifier(pathSegments.toArray(new String[pathSegments.size()]));
```

Comment:
why don't we simply pass in the *path* variable?

Reply:
Right, I've changed it now.

Reply:
There seem to be some unused imports remaining ..